File: MainListener.js
const antlr4 = require('antlr4/index');
const Lexer = require('./VectorsLexer');
const Parser = require('./VectorsParser');
let VectorsListener = require('./VectorsListener').VectorsListener;
 
MainListener = function() {
    this.table = {};
    VectorsListener.call(this);
    return this;
};
 
MainListener.prototype = Object.create(VectorsListener.prototype);
MainListener.prototype.constructor = MainListener;
 
MainListener.prototype.enterStatement = function(ctx) {
    // PRINT ID
    if(ctx.PRINT()){
        console.log(this.table[ctx.children[1].getText()]);
    } 
};

MainListener.prototype.exitDeclaration = function(ctx) {
    let id = ctx.ID().getText();
    // ID AO e
    if(ctx.children.length===3){
        if(ctx.children[2].ID()!=null){
            // ID AO ID
            let tmp = this.table[ctx.children[2].ID().getText()];
            this.table[id] = tmp.slice();
        }else{
            // ID AO e
            let tmp = JSON.parse(ctx.children[2].getText());
            this.table[id] = tmp.slice();
        }
    // ID
    }else{
        this.table[id] = [0,0,0];
    }
}; 

// ID AO e
MainListener.prototype.exitAssignment = MainListener.prototype.exitDeclaration;

MainListener.prototype.exitE = function(ctx) {
    // ID
    if(ctx.ID()){
        id=ctx.ID().getText();
        return this.table[id].slice();
    // LP NUM COMMA NUM COMMA NUM DP 
    }else{
        let tmp=[];
        for(let i=1;i<=5;i+=2){
            tmp.push(ctx.children[i].getText());
        }
        return tmp;
    }
};

exports.MainListener = MainListener;

File: Python/MainListener.py
import json
from antlr4 import *
from VectorsParser import VectorsParser
from VectorsListener import VectorsListener

class MainListener(VectorsListener):
    def __init__(self):
        self.table = {}
    
    def exitStatement(self, ctx):
        ## PRINT ID
        if(ctx.PRINT()):
            print(self.table[ctx.children[1].getText()])

    def exitE(self, ctx):
        ## ID
        if(ctx.ID()):
            id = ctx.ID().getText()
            return self.table[id]
        ## LP NUM COMMA NUM COMMA NUM RP
        else:
            tmp = []
            for x in range(1,6)[::2]:
                tmp.append(ctx.children[x].getText())
            return tmp

    def exitDeclaration(self, ctx):
        id = ctx.ID().getText()
        if(len(ctx.children)==3):
            ##if(ctx.children[2].ID() is not None):
                ##tmp1 = self.table[ctx.children[2].ID().getText()]
                ##self.table[id] = tmp1
            ##else:
            ## ID AO e
            tmp2 = json.loads(ctx.children[2].getText())
            self.table[id] = tmp2
        else:
            ## ID
            self.table[id] = []

    def exitAssignment(self, ctx):
        id = ctx.ID().getText()
        ## ID AO e
        if(len(ctx.children)==3):
            if(ctx.children[2].ID() is not None):
                tmp1 = self.table[ctx.children[2].ID().getText()]
                self.table[id] = tmp1
            else:
                tmp2 = json.loads(ctx.children[2].getText())
                self.table[id] = tmp2
        ##else:
            ##self.table[id] = []
